src/components/ReservationForm.jsx

```diff
@@ -1,9 +1,88 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { postReservation } from 'redux/reservations/carReserve';
+import { Link } from 'react-router-dom';
 
-const ReservationForm = () => (
-  <div>
-    use for making reservations
-  </div>
-);
+const ReservationForm = () => {
+  const { reservation: { isLoading }, cars: { cars } } = useSelector((store) => store);
+  const dispatch = useDispatch();
+  const [city, setCity] = useState('');
+  const [reservationDate, setReservationDate] = useState('');
+  const [returningDate, setReturningDate] = useState('');
+  const [carId, setCarId] = useState('');
+  const [error, setError] = useState('');
+  const handleReservation = async (event) => {
+    event.preventDefault();
+    try {
+      await dispatch(postReservation({
+        carId, city, reservationDate, returningDate,
+      }));
+    } catch (error) {
+      setError(error.message);
+    }
+  };
+  return (
+    <div className="form-wrap">
+      <h3>Reserve A Car</h3>
+      <form onSubmit={handleReservation}>
+        {error && <p>{error}</p>}
+        <label className="form-label" htmlFor="city">
+          Enter your city
+          <input
+            className="form-control"
+            type="text"
+            id="city"
+            value={city}
+            onChange={(e) => setCity(e.target.value)}
+            placeholder="Enter Your City"
+          />
+        </label>
```

Comment:
**To reserve an appointment, the user has to select a date and city (username and selected item are autofilled).
Use the design based on the ["Book a vespa test-ride"](https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign/modules/173005583) and add all necessary inputs.
The user can also access the "Reserve" page from the navigation panel. In that case only username is autofilled.**

Based on the above project requirement bove, the **city** field in the form should be  **select** from a list of existing cities instead of an **input** of **type="text"**